src/main/java/WayofTime/bloodmagic/ritual/types/RitualEternalSoul.java

```diff
@@ -0,0 +1,127 @@
+import WayofTime.bloodmagic.BloodMagic;
+import WayofTime.bloodmagic.altar.IBloodAltar;
+import WayofTime.bloodmagic.block.BlockLifeEssence;
+import WayofTime.bloodmagic.ritual.*;
+import WayofTime.bloodmagic.util.helper.NetworkHelper;
+import WayofTime.bloodmagic.util.helper.PlayerHelper;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+@RitualRegister("eternal_soul")
+public class RitualEternalSoul extends Ritual {
+
+    private UUID owner;
+    private int currentEssence;
+    private World world;
+    private BlockPos pos;
+    private IBloodAltar altar = null;
+    private EntityPlayer entityOwner;
+
+    public RitualEternalSoul() {
+        super("ritualEternalSoul", 2, 2000000, "ritual." + BloodMagic.MODID + ".eternalSoulRitual");
+    }
+
+    @Override
+    public void performRitual(IMasterRitualStone masterRitualStone) {
+        this.owner = masterRitualStone.getOwner();
+        this.currentEssence = NetworkHelper.getSoulNetwork(this.owner).getCurrentEssence();
+        this.world = masterRitualStone.getWorldObj();
+        this.pos = masterRitualStone.getBlockPos();
+
+        if (this.altar == null) {
+            for (int i = -5; i <= 5; i++) {
+                for (int j = -5; j <= 5; j++) {
+                    for (int k = -10; k <= 10; k++) {
+                        if (this.world.getTileEntity(new BlockPos(pos.getX() + i, pos.getY() + j, pos.getZ() + k)) instanceof IBloodAltar) {
+                            this.altar = (IBloodAltar) this.world.getTileEntity(new BlockPos(pos.getX() + i, pos.getY() + j, pos.getZ() + k));
+                        }
+                    }
+                }
+            }
+        }
+        if (!(this.altar instanceof IFluidHandler))
+            return;
+
+        int horizontalRange = 15;
+        int verticalRange = 20;
+
+        List<EntityPlayer> list = this.world.getEntitiesWithinAABB(EntityPlayer.class,
+                new AxisAlignedBB(pos.getX() - 0.5f, pos.getY() - 0.5f, pos.getZ() - 0.5f,
+                        pos.getX() + 0.5f, pos.getY() + 0.5f, pos.getZ() + 0.5f)
+                        .expand(horizontalRange, verticalRange, horizontalRange));
+
+        for (EntityPlayer player : list) {
+            if (PlayerHelper.getUUIDFromPlayer(player) == this.owner)
+                this.entityOwner = player;
+        }
+
+        int fillAmount = Math.min(this.currentEssence / 2, ((IFluidHandler) this.altar).fill(new FluidStack(BlockLifeEssence.getLifeEssence(), 10000), false));
+
+        ((IFluidHandler) this.altar).fill(new FluidStack(BlockLifeEssence.getLifeEssence(), fillAmount), true);
+
+        if (this.entityOwner.getHealth() > 2.0f && fillAmount != 0)
```

Comment:
What's this doing?

Reply:
It fills the altar to the maximum (it should be the maximum buffer) or the remaining essence in the network divided by 2 (because the other half is transfer tax).

While the ritual takes effect, the owners health is reduced to 2 as per 1.7.10.

Reply:
Oh, I totally mixed this up with the condor when I was going over, oops. Yeah setting health makes sense I guess?

Reply:
I have no idea, I just ported it.

Reply:
prevents abuse by storing up LP and then using it in conjunction with sacrifices, I guess

Reply:
However, if it only fills up the buffer, it's basically the same as filling the altar up with a tank (which are easily to come by since BM2 and in almost every modpack anyways)